.ci/script.sh

```diff
@@ -0,0 +1,79 @@
+#!/usr/bin/env bash
+set -uo pipefail
+
+echo "which java: $(which java)"
+ulimit -c unlimited -S
+
+gradle build -x :web:test --no-daemon --stacktrace --info
+
+# Print core dumps when JVM crashes.
+RESULT=$?
+
+if [[ ${RESULT} -ne 0 ]]; then
+  ls "$TRAVIS_BUILD_DIR"/images
+  zip --verbose -r "$TRAVIS_BUILD_DIR"/images.zip "$TRAVIS_BUILD_DIR"/images/*
+
+  if [ ! -z ${PRINT_CRASH_LOGS+x} ]; then
+    if [[ "${TRAVIS_OS_NAME}" == osx ]]; then FIND="gfind"; else FIND="find"; fi
+    ${FIND} . -name "hs_err_pid*.log" -type f -printf '\n====== JVM CRASH LOG ======\n%p\n' -exec cat {} \;
+
+    CORES=''
+    if [[ "${TRAVIS_OS_NAME}" == osx ]]; then
+      CORES="$(find /cores -type f -regex '*core.[0-9]{4}' -print)"
+    else
+      CORES="$(find . -type f -regex '*core.[0-9]{6}' -print)"
+    fi
+
+    if [ -n "${CORES}" ]; then
+      for core in ${CORES}; do
+        printf '\n\n======= Core file %s =======\n' "$core"
+        if [[ "${TRAVIS_OS_NAME}" == osx ]]; then
+          lldb -Q -o "bt all" -f "$(which java)" -c "${core}"
+        else
+          gdb -n -batch -ex "thread apply all bt" -ex "set pagination 0" "$(which java)" -c "${core}"
+        fi
+      done
+    fi
+  fi
+  exit ${RESULT}
+  # exit 0 # Force success so images are uploaded.
+fi
+
+# Download JDK 10 to use for boot JDK for JDK 11.
+cd "$TRAVIS_BUILD_DIR"
+if [[ "${TRAVIS_OS_NAME}" == osx ]]; then
+  ORACLE_JDK10='jdk-10-ea+42_osx-x64_bin.dmg'
+  wget https://download.java.net/java/jdk10/archive/42/BCL/"$ORACLE_JDK10"
```

Comment:
Could you define a BOOT_JDK variable to define which jdk file (prefix) to download ?
Why do you need ORACLE jdk instead of OpenJDK EA builds ? (need jfx modules ?)

Reply:
Using OpenJDK as the boot JDK I got errors when running the OpenJDK configure script that said the boot JDK was the wrong version. The errors did not happen with Oracle JDK.

Reply:
I mean use a more generic variable name and possibly move BOOT_JDK_MAC and BOOT_JDK_LINUX at the top of the script.
Ideally these could share the 'jdk-10-ea+42' prefix.
That's good enough and can be made later

Reply:
Okay, I did something very close to that, where it is easy to change the JDK10 version at the top of the script.